src/components/top-bar-equipment-seach-dialog/custom-suffix-renderer.tsx

```diff
@@ -76,19 +74,18 @@ export const CustomSuffixRenderer: FunctionComponent<
                 e.stopPropagation();
             });
         },
-        [dispatch, onClose, studyUuid, currentNode]
+        [studyUuid, currentNode, element.type, element.id, dispatch, onClose]
     );
 
-    const openNetworkAreaDiagramCB = useCallback(
-        (
-            e: React.MouseEvent<HTMLButtonElement, MouseEvent>,
-            element: EquipmentInfos
-        ) => {
+    const openNetworkAreaDiagramCB = useCallback<
+        MouseEventHandler<HTMLButtonElement>
+    >(
+        (e) => {
```

Comment:
```suggestion
    const openNetworkAreaDiagramCB = useCallback(
        (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
```
I have no type checking error with the code above

Reply:
the error happen on the "IconButton.onClick":
> TS2345: Argument of type MouseEvent<HTMLButtonElement, MouseEvent> is not assignable to parameter of type
>> MouseEvent<HTMLButtonElement, MouseEvent<Element, MouseEvent>>